quickstart/101-vm-with-infrastructure/main.tf

```diff
@@ -116,13 +110,12 @@ resource "azurerm_linux_virtual_machine" "my_terraform_vm" {
     version   = "latest"
   }
 
-  computer_name                   = "myvm"
-  admin_username                  = "azureuser"
-  disable_password_authentication = true
+  computer_name                   = "hostname"
+  admin_username                  = coalesce(var.username, "azureuser")
 
   admin_ssh_key {
-    username   = "azureuser"
-    public_key = tls_private_key.example_ssh.public_key_openssh
+    username   = coalesce(var.username, "azureuser")
```

Comment:
Same issue here.

Reply:
Using the coalesce() function allows us to give the reader two options: variable or random. If the reader blanks out the var, they get a random value - such as if they're testing the code by running it multiple times and can't use the same value twice.